src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmController.java

```diff
@@ -1,211 +1,182 @@
+
+    @ApiOperation(value = "Updates a list of Case Documents with provided Metadata")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Created", response = StoredDocumentHalResource.class),
+        @ApiResponse(code = 204, message = "No Content")})
+    @RequestMapping(value = "/cases/documents/attachToCase",
+        produces = {APPLICATION_JSON},
+        consumes = {APPLICATION_JSON},
+        method = RequestMethod.PATCH)
+    public ResponseEntity<Object> patchMetaDataOnDocuments(
         @ApiParam(value = "", required = true)
         @Valid @RequestBody DocumentMetadata caseDocumentMetadata,
-
-        @ApiParam(value = "Service Auth (S2S). Use it when accessing the API on App Tier level.", required = true)
-        @RequestHeader(value = "ServiceAuthorization", required = true) String serviceAuthorization,
-
-        @ApiParam("Authorization header of the currently authenticated user")
-        @RequestHeader(value = "Authorization", required = true) String authorization,
-
-        @ApiParam("User-Id of the currently authenticated user. If provided will be used to populate the creator field of a document"
-            + " and will be used for authorisation.")
-        @RequestHeader(value = "user-id", required = false) String userId,
-
-        @ApiParam("Comma-separated list of roles of the currently authenticated user. If provided will be used for authorisation.")
-        @RequestHeader(value = "user-roles", required = false) String userRoles) {
+        @RequestHeader(value = "user-id", required = false) String userId) {
 
         try {
-            if (!ValidationService.validate(caseDocumentMetadata.getCaseId())) {
-                throw new BadRequestException("The Case Id is invalid");
-            }
-            ValidationService.validateInputParams(INPUT_CASE_ID_PATTERN, caseDocumentMetadata.getCaseId());
-            caseDocumentMetadata.getDocuments()
-                                .forEach(document -> {
-                                    ValidationService.validateInputParams(INPUT_STRING_PATTERN, document.getId());
-                                    ValidationService.validateDocumentId(document.getId());
-                                });
-
-            documentManagementService.patchDocumentMetadata(caseDocumentMetadata, serviceAuthorization, userId);
-        } catch (BadRequestException | IllegalArgumentException e) {
-            throw new BadRequestException("Exception while attaching the documents to a case :" + e);
+            documentManagementService.patchDocumentMetadata(caseDocumentMetadata, userId);
         } catch (Exception e) {
-            LOG.error("Exception in controller for patch MetaData Documents API");
+            LOG.error("Exception while attaching the documents to a case :" + e);
             throw e;
         }
-        return new ResponseEntity<>(HttpStatus.OK);
+        return new ResponseEntity<Object>(HttpStatus.OK);
```

Comment:
This object type was also removed because of sonar smell. we can check rom Nitin.

Reply:
Done.